### ISSUE TITLE
Understanding the coherence of the training

### ISSUE THREAD

**Issue by tpereztorres** [AlexeyAB/darknet#2133]
Hello @AlexeyAB,
 
We would like to ask you two doubts about training experience.

First, we have found a problem with the performance during the training. For example, the precision is better at the step 12.000 but it is worse at the step 50.000. However, the best precision is obtained at the step 120.000.

Our dataset contains 65.000 images with more than one bounding box in each image of only one class.

We understand the meaning of overfitting but we do not find the coherence with the obtaining results. Why the training improves and get worse with so much dynamism?

Another doubt is that when we reduce the depth of the network, is it correct that in order to obtain the same or similar best precision (of the step 120.000) we need to train more epochs?

Thank you very much.

**Comment by AlexeyAB** [comment]
@tpereztorres Hello,

1. What mAP did you get at 12000, 50000, 120000 iterations?

2. Did you measure mAP for separate validation dataset? How many images in the validation dataset?

3. Did you use yolov3.cfg or yolov3-tiny.cfg?

4. Can you show `width, height, batch, subdivisions, learning_rate, scales, steps, random, jitter` from your cfg-file?

**Comment by AlexeyAB** [comment]
> We can not specifiy the values of some metrics as mAP because we test the weights in real time for a specific application.

What accuracy metrics do you use?

>  For example, the precision is better at the step 12.000 but it is worse at the step 50.000. However, the best precision is obtained at the step 120.000.

How did you measure precision? What dataset do you use for this?


> We understand the meaning of overfitting but we do not find the coherence with the obtaining results. Why the training improves and get worse with so much dynamism?

What is the value of fluctuations of accuracy in percent?

> Another doubt is that when we reduce the depth of the network, is it correct that in order to obtain the same or similar best precision (of the step 120.000) we need to train more epochs?

Precisely speaking, **deeper network with Residual connection has smaller Error** for the same iteration number. Page 8, Figure 6 https://arxiv.org/pdf/1512.03385.pdf
![image](https://user-images.githubusercontent.com/4096485/50687275-b35a6b00-1031-11e9-8759-121c91c09808.png)


----

To get higher accuracy - better to use:
* deeper network with residual connections
* lower size of filters
* lower number of filters
* in the intermediate layers better to use 32 768 filters (bit-1) intead of 1024 filters (float-32-bit)
I.e.
    * several float-32 layers
    * several int-8 layers
    * several int-4 layers
    * several bit-1 layers
    * several int-4 layers
    * several int-8 layers

**Comment by tpereztorres** [comment]
Hello again @AlexeyAB,

Thank you for the quick answer.

Just one question, do you mean to use a number of filters between 32 and 768 instead of 1024?

Thank you very much.
